flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/FlinkService.java

```diff
@@ -136,9 +136,20 @@ public boolean isJobManagerPortReady(Configuration config) {
         return false;
     }
 
+    public boolean isJobManagerServing(Configuration config) {
+        try {
+            this.listJobs(config);
+            return true;
+        } catch (Exception ignored) {
+            return false;
+        }
+    }
+
     public Collection<JobStatusMessage> listJobs(Configuration conf) throws Exception {
         try (ClusterClient<String> clusterClient = getClusterClient(conf)) {
-            return clusterClient.listJobs().get(10, TimeUnit.SECONDS);
+            return clusterClient
+                    .listJobs()
+                    .get(operatorConfiguration.getProgressCheckIntervalSeconds(), TimeUnit.SECONDS);
```

Comment:
I think we need a new config in the OperatorConfiguraiton for this. We could call it `flinkClientTimeoutSeconds` -> `flink.client.timeout.seconds`

Reply:
updated.